src/internal/scheduler/animationFrame.ts

```diff
@@ -71,27 +84,60 @@ export const animationFrame: SchedulerLike = {
   now: DEFAULT_NOW,
 };
 
-function flushQueue() {
-  try {
-    while (_queue.length > 0) {
-      const work = _queue.shift() as (state: any) => void;
-      const state = _queue.shift();
-      const subscription = _queue.shift() as Subscription;
-      try {
-        work(state);
-      } finally {
-        subscription.unsubscribe();
-      }
-    }
-  } finally {
-    // clean up if necessary
-    if (_queue.length > 0) {
-      const copy = _queue.slice();
-      _queue.length = 0;
-      for (let i = 2; i < copy.length; i += 3) {
-        (copy[i] as Subscription).unsubscribe();
+let flushing = false;
+
+/** Moves to the next animation task queue and flushes it */
+function flush() {
+  if (!flushing) {
+    flushing = true;
+
+    // Move to the next queue
+    _currentQueue = _nextQueue;
+
+    // Start a new queue for newly scheduled or rescheduled items
+    _nextQueue = [];
+
+    // process the tasks on this queue FIFO.
+    while (_currentQueue.length > 0) {
+      const task = _currentQueue.shift();
```

Comment:
(continuing the comment thread) could this be a non-destructive `for (let i = 0; i++ < currentQueueLength;) {}` loop rather than `shift()`?